huxley/core/models.py

```diff
@@ -266,53 +266,56 @@ def update_assignments(cls, new_assignments):
 
         def add(committee, country, school, rejected):
             additions.append(cls(
-                committee_id=committee,
-                country_id=country,
-                school_id=school,
+                committee_id=committee.id,
+                country_id=country.id,
+                school_id=school.id,
                 rejected=rejected,
             ))
 
         def remove(assignment_data):
             deletions.append(assignment_data['id'])
 
         for committee, country, school, rejected in new_assignments:
-            # If one or more fields in the assignment are invalid, the first 
-            # three values are passed in as strings instead of ints
-            is_valid = type(committee) is int
-
             key = (committee, country)
             if key in assigned:
                 # Make sure that the same committee/country pair is not being
                 # given to more than one school in the upload
-                committee = str(Committee.objects.get(pk=committee).name) + ' - DUPLICATE ASSIGNMENT IN CSV'
-                country = str(Country.objects.get(pk=country).name) + ' - DUPLICATE ASSIGNMENT IN CSV'
-                school = str(School.objects.get(pk=school).name)
+                committee = str(committee.name) + ' - DUPLICATE ASSIGNMENT IN CSV'
+                country = str(country.name) + ' - DUPLICATE ASSIGNMENT IN CSV'
+                school = str(school.name)
                 failed_assignments.append(str((school, committee, country, rejected)))
                 continue
 
-            if is_valid:
-                assigned.add(key)
-                old_assignment = assignment_dict.get(key)
+            # If the assignemnt contains no bad cells, then each value should
+            # have the type of its corresponding model.
+            is_valid = True
+            if type(committee) is not Committee:
+                committee = Committee(name=str(committee)+' - DOES NOT EXIST')
```

Comment:
I did this to improve style consistency in the output. If a Committee is created then I can use the name field without checking the type of the committee, i.e. otherwise to keep the output in a consistent format I would have to convert the committee to a string before saving it which would clutter the code a lot more.